src/interface/make_pack_descriptor.hpp

```diff
@@ -136,6 +137,16 @@ inline auto MakePackDescriptor(StateDescriptor *psd, const std::vector<Uid_t> &v
   return typename SparsePack<>::Descriptor(base_desc);
 }
 
+template <class... Types, class... Args>
+inline auto MakePackDescriptorFromTypeList(TypeList<Types...>, Args &&...args) {
+  return MakePackDescriptor<Types...>(std::forward<Args>(args)...);
+}
+
+template <class TL, class... Args>
+inline auto MakePackDescriptorFromTypeList(Args &&...args) {
+  return MakePackDescriptorFromTypeList(TL(), std::forward<Args>(args)...);
+}
+
```

Comment:
```suggestion
template <template <class...> class TL, class... Types, class... Args>
inline auto MakePackDescriptorFromTypeList(TL<Types...>, Args &&...args) {
  return MakePackDescriptor<Types...>(std::forward<Args>(args)...);
}

template <class TL, class... Args>
inline auto MakePackDescriptorFromTypeList(Args &&...args) {
  return MakePackDescriptorFromTypeList(TL(), std::forward<Args>(args)...);
}
```
Would a template template parameter like this work? It could provide more flexibility for a downstream code to use their own type lists

Reply:
yeah, that is a nice way to do it. suggestion committed.